### ISSUE TITLE
Improved GUI for "image type" in dynamic profile rules

### ISSUE THREAD

**Issue by agriggio** [Beep6581/RawTherapee#4901]
I think a combo box works better for specifying image types (STD, HDR, or PS) in dynamic profile rules. Here's a patch that does that. Screenshot:

![screenshot from 2018-10-29 09 35 32](https://user-images.githubusercontent.com/11982086/47638242-496ba400-db5e-11e8-9c7f-e2b96ea91fc9.jpeg)

```diff
diff --git a/rtdata/languages/default b/rtdata/languages/default
--- a/rtdata/languages/default
+++ b/rtdata/languages/default
@@ -43,6 +43,10 @@
 DYNPROFILEEDITOR_EDIT;Edit
 DYNPROFILEEDITOR_EDIT_RULE;Edit Dynamic Profile Rule
 DYNPROFILEEDITOR_ENTRY_TOOLTIP;The matching is case insensitive.\nUse the "re:" prefix to enter\na regular expression.
+DYNPROFILEEDITOR_IMGTYPE_ANY;Any
+DYNPROFILEEDITOR_IMGTYPE_STD;Standard
+DYNPROFILEEDITOR_IMGTYPE_HDR;HDR
+DYNPROFILEEDITOR_IMGTYPE_PS;Pixel Shift
 DYNPROFILEEDITOR_MOVE_DOWN;Move Down
 DYNPROFILEEDITOR_MOVE_UP;Move Up
 DYNPROFILEEDITOR_NEW;New
diff --git a/rtgui/dynamicprofilepanel.cc b/rtgui/dynamicprofilepanel.cc
--- a/rtgui/dynamicprofilepanel.cc
+++ b/rtgui/dynamicprofilepanel.cc
@@ -41,7 +41,17 @@
 
     add_optional (M ("EXIFFILTER_CAMERA"), has_camera_, camera_);
     add_optional (M ("EXIFFILTER_LENS"), has_lens_, lens_);
-    add_optional (M ("EXIFFILTER_IMAGETYPE"), has_imagetype_, imagetype_);
+    
+    imagetype_ = Gtk::manage (new MyComboBoxText());
+    imagetype_->append(Glib::ustring("(") + M("DYNPROFILEEDITOR_IMGTYPE_ANY") + ")");
+    imagetype_->append(M("DYNPROFILEEDITOR_IMGTYPE_STD"));
+    imagetype_->append(M("DYNPROFILEEDITOR_IMGTYPE_HDR"));
+    imagetype_->append(M("DYNPROFILEEDITOR_IMGTYPE_PS"));
+    imagetype_->set_active(0);
+    hb = Gtk::manage (new Gtk::HBox());
+    hb->pack_start (*Gtk::manage (new Gtk::Label (M ("EXIFFILTER_IMAGETYPE"))), false, false, 4);
+    hb->pack_start (*imagetype_, true, true, 2);
+    get_content_area()->pack_start (*hb, Gtk::PACK_SHRINK, 4);
 
     add_range (M ("EXIFFILTER_ISO"), iso_min_, iso_max_);
     add_range (M ("EXIFFILTER_APERTURE"), fnumber_min_, fnumber_max_);
@@ -82,8 +92,17 @@
     has_lens_->set_active (rule.lens.enabled);
     lens_->set_text (rule.lens.value);
 
-    has_imagetype_->set_active (rule.imagetype.enabled);
-    imagetype_->set_text (rule.imagetype.value);
+    if (!rule.imagetype.enabled) {
+        imagetype_->set_active(0);
+    } else if (rule.imagetype.value == "STD") {
+        imagetype_->set_active(1);
+    } else if (rule.imagetype.value == "HDR") {
+        imagetype_->set_active(2);
+    } else if (rule.imagetype.value == "PS") {
+        imagetype_->set_active(3);
+    } else {
+        imagetype_->set_active(0);
+    }
 
     profilepath_->updateProfileList();
 
@@ -116,8 +135,20 @@
     ret.lens.enabled = has_lens_->get_active();
     ret.lens.value = lens_->get_text();
 
-    ret.imagetype.enabled = has_imagetype_->get_active();
-    ret.imagetype.value = imagetype_->get_text();
+    ret.imagetype.enabled = imagetype_->get_active_row_number() > 0;
+    switch (imagetype_->get_active_row_number()) {
+    case 1:
+        ret.imagetype.value = "STD";
+        break;
+    case 2:
+        ret.imagetype.value = "HDR";
+        break;
+    case 3:
+        ret.imagetype.value = "PS";
+        break;
+    default:
+        ret.imagetype.value = "";
+    }
 
     ret.profilepath = profilepath_->getFullPathFromActiveRow();
 
@@ -478,7 +509,14 @@
 void DynamicProfilePanel::render_imagetype (
     Gtk::CellRenderer *cell, const Gtk::TreeModel::iterator &iter)
 {
-    RENDER_OPTIONAL_ (imagetype);
+    auto row = *iter;
+    Gtk::CellRendererText *ct = static_cast<Gtk::CellRendererText *>(cell);
+    DynamicProfileRule::Optional o = row[columns_.imagetype];
+    if (o.enabled) {
+        ct->property_text() = M(std::string("DYNPROFILEEDITOR_IMGTYPE_") + o.value);
+    } else { \
+        ct->property_text() = "";
+    }
 }
 
 #undef RENDER_OPTIONAL_
diff --git a/rtgui/dynamicprofilepanel.h b/rtgui/dynamicprofilepanel.h
--- a/rtgui/dynamicprofilepanel.h
+++ b/rtgui/dynamicprofilepanel.h
@@ -112,8 +112,7 @@
         Gtk::CheckButton *has_lens_;
         Gtk::Entry *lens_;
 
-        Gtk::CheckButton *has_imagetype_;
-        Gtk::Entry *imagetype_;
+        MyComboBoxText *imagetype_;
 
         ProfileStoreComboBox *profilepath_;
     };
```

**Comment by Beep6581** [comment]
@agriggio would a catch-all "raw" and "non-raw" filter be possible? That's with regard to #4488

**Comment by agriggio** [comment]
I think it should be possible, yes. but how should it be combined with the current image types?

**Comment by heckflosse** [comment]
@agriggio Alberto, :+1: to push the current state of your work

**Comment by Beep6581** [comment]
@agriggio I haven't replied yet because I haven't come to a conclusion yet which is better:
* two logical-AND combobox choices, one for raw/non-raw/either and one for any/standard/HDR/PS
* one choice consisting of:
  * Any non-raw
  * HDR non-raw
  * Any raw
  * Standard raw
  * HDR raw
  * PS

So far two comboboxes sounds simpler, even though some combinations might not be encountered, such as "non-raw AND Pixel Shift".

**Comment by agriggio** [comment]
honestly, I think two combos are overkill, and even a single one with all the combinations is too much IMHO. In fact, I don't have a strong use-case for dynamic profiles for non-raw images, as in the (rare) cases in which I use RT for non-raw, I always start from neutral. But I do understand there's some need for it, so... I don't know what to do either :slightly_frowning_face:

**Comment by heckflosse** [comment]
@agriggio Let's merge your branch as is now and think about it again for 5.6

**Comment by Beep6581** [comment]
The use case for filtering by non-raw images is #4488 but as that issue is for 5.6, I agree with @heckflosse that we can merge this and expand on it later.

**Comment by Beep6581** [comment]
We can throw out "HDR non-raw", which leaves us with 5 useful options.

**Comment by agriggio** [comment]
I think 5 is ok, but ok for postponing to 5.6. I'll merge later

**Comment by agriggio** [comment]
Implemented in 4d9871110bd47bb8399f204d6f6ba75c6267639e
